util-images/access-tokens/cmd/main.go

```diff
@@ -29,18 +29,21 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	certutil "k8s.io/client-go/util/cert"
+	flowcontrol "k8s.io/client-go/util/flowcontrol"
 	"k8s.io/klog"
 )
 
 var (
-	namespace = flag.String("namespace", "", "Namespace that is targeted with request flow.")
-	tokenDirs = flag.StringSlice("access-token-dirs", nil, "Directories with access-token. A worker will be started per each dir/token")
-	qps       = flag.Float64("qps-per-worker", 0.5, "QPS per worker")
+	namespace       = flag.String("namespace", "", "Namespace that is targeted with request flow.")
+	tokenDirs       = flag.StringSlice("access-token-dirs", nil, "Directories with access-token. A worker will be started per each dir/token")
+	qps             = flag.Float64("qps-per-worker", 0.5, "QPS per worker")
+	workersPerToken = flag.Int("workers-per-token", 1, "Number of workers using the same token")
```

Comment:
Why do we need this additional layer here of parallelism here? Is there any use-case for it?

The downside of it is that it's getting more complicated, total qps is now: #tokens * qps * worksersPerToken

Reply:
It was mostly related to broken implementation of `func worker` that could be throttled in a case if server in not responding fast enough. Changed implementation to asynchronous and removed `workersPerToken` flag